### DISCUSSION TITLE
Stack in app template?

### DISCUSSION THREAD

**Discussion by awcodes** [filamentphp/filament#1778] [Ideas]
Would it be out of place to ask for a stack in the app.blade.php layout before the closing tag?

I have a need to push a livewire component to the page but currently cannot do that without publishing the the file locally, which it less than desired in order to maintain filament’s views/layouts with future releases.

The only other stack I’ve found is the scripts stack but that’s outside of the layout and livewire can’t pick it up.

**Comment by danharrin** [comment]
I've got nothing against it if you can think of a suitable & non-confusing name... which is the hard bit.

**Reply by awcodes** [reply]
Yea. Naming is hard. In my use case it’s for a custom modal for a field, but there is already a modal $variable. But I can also see application for it on page and list views too.

Maybe something like ‘page-support’? Personally not a fan of multi word stack names though.

**Reply by danharrin** [reply]
Which component are you pushing?

**Reply by awcodes** [reply]
It a custom modal that is mostly full width. Trigger's from a button in a field component.

![Screen Shot 2022-03-03 at 5 04 26 PM](https://user-images.githubusercontent.com/3596800/156792628-3b0b1b56-bad1-4fb7-adc8-9dd80f663402.png)

![Screen Shot 2022-03-03 at 5 04 19 PM](https://user-images.githubusercontent.com/3596800/156792638-4cd7e987-a967-408d-8ef4-c63d73b1f272.png)

![Screen Shot 2022-03-03 at 5 04 31 PM](https://user-images.githubusercontent.com/3596800/156792645-2f70d8e9-4a3f-4a6d-9ba4-62a25cabb0d6.png)

[Here's my MVP](https://github.com/awcodes/trov-filament/tree/main/packages/trov-media-library) so far, if you'd care to have a look.

**Reply by danharrin** [reply]
Fair enough, how about we use an `modals` stack then on the `<x-page>` Blade component? If it's int the layout, then Livewire will not be able to dynamically push to it after the page is first loaded, so your field wont work on relation managers or in other modals.

**Reply by awcodes** [reply]
That sounds good to me. I’m dynamically pushing it from the service provider so it only loads once and works in repeaters and blocks.

Thank for the discussion and feed back.

Would you prefer to handle this in a future release or would it be better if I did a PR?

**Reply by danharrin** [reply]
You shouldn't need to do it in a service provider tbh, using `@once @push('modals')` in your form component view should work.

**Reply by awcodes** [reply]
Can you expound on that a little. My component extends the field component. If I push from there it won’t work in a repeater or builder block since it wouldn’t be rendered until it was added to the repeater or builder block. That’s why I did it from the provider. I don’t like having it on every page but it seemed to me it was the only way to guarantee the modal was on the page to listen for the dispatch from the form field if there was no field initially on the page to do the push.

**Reply by danharrin** [reply]
I suggested that we add the stack to the `<x-filament::page>` component and not the layout, since this is part of the Livewire view and will be rerendered between requests. So your component will work on repeaters and builders :)

**Reply by awcodes** [reply]
I’m ok with that. Thank you again.

**Reply by danharrin** [reply]
Alight, lmk when you have a PR submitted and I'll merge it.

**Comment by awcodes** [comment]
Thanks again @danharrin PR has been submitted.
